summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```diff
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Comment:
Yes, I meant usage of `flatMap` in future, for now I'm ok with `match` and inclusive/exclusive intervals return.